vault/resource_rabbitmq_secret_backend_role.go

```diff
@@ -193,3 +204,108 @@ func rabbitmqSecretBackendRoleExists(d *schema.ResourceData, meta interface{}) (
 	log.Printf("[DEBUG] Checked if %q exists", path)
 	return secret != nil, nil
 }
+
+func expandRabbitmqSecretBackendRoleVhost(vhost []interface{}, typ string) (string, map[string]interface{}, error) {
+	log.Printf("[DEBUG] Vhosts as list from ResourceData: %+v", vhost)
+
+	vhosts := make(map[string]interface{}, len(vhost))
+
+	for _, host := range vhost {
+		h := map[string]interface{}{}
+		var id string
+		for k, v := range host.(map[string]interface{}) {
+			if k == typ {
+				id = v.(string)
+				continue
+			}
+			h[k] = v
+		}
+		vhosts[id] = h
```

Comment:
I think before setting the vhosts here we should confirm that `id != ""`. If no match was found and `id == ""` we should return an error, since otherwise we'd map our vhost permissions to `""`

Reply:
done